sdk/src/Core/Amazon.Runtime/Pipeline/ErrorHandler/HttpErrorResponseExceptionHandler.cs

```diff
@@ -115,6 +115,87 @@ public override bool HandleException(IExecutionContext executionContext, HttpErr
             throw errorResponseException;
         }
 
+#if AWS_ASYNC_API
+        /// <summary>
+        /// Handles an exception for the given execution context.
+        /// </summary>
+        /// <param name="executionContext">The execution context, it contains the
+        /// request and response context.</param>
+        /// <param name="exception">The exception to handle.</param>
+        /// <returns>
+        /// Returns a boolean value which indicates if the original exception
+        /// should be rethrown.
+        /// This method can also throw a new exception to replace the original exception.
+        /// </returns>
+        public override async System.Threading.Tasks.Task<bool> HandleExceptionAsync(IExecutionContext executionContext, HttpErrorResponseException exception)
```

Comment:
We need to refactor this and HandleException to not have so much duplicate code.